provisioning/tasks/hostname.yml

```diff
@@ -0,0 +1,28 @@
+- name: Define fully qualified domain name.
+  set_fact:
+    hostname_fqdn: "{{ vagrant_hostname }}"
+  when: hostname_fqdn is undefined
+
+- name: Define short hostname.
+  set_fact:
+    hostname_short: "{{ hostname_fqdn|regex_replace('^([^.]+).*$', '\\1') }}"
+  when: hostname_short is undefined
+
+- name: Add hostname to /etc/hosts.
+  lineinfile:
+    dest: /etc/hosts
+    regexp: '.*\t{{ hostname_short }}$'
+    line: "127.0.0.1\t{{ hostname_fqdn }}\t{{ hostname_short }}"
+    state: present
+    unsafe_writes: "{{ (ansible_virtualization_type == 'docker')|ternary('yes', 'no')|bool }}"
+
+- name: Configure hostname.
+  copy:
+    content: "{{ (ansible_os_family == 'Debian') | ternary(hostname_short, hostname_fqdn) }}\n"
+    dest: /etc/hostname
+    unsafe_writes: "{{ (ansible_virtualization_type == 'docker')|ternary('yes', 'no')|bool }}"
```

Comment:
Also note that this will be the 2nd thing (besides `systemd` use in the Ansible role) that explicitly requires Ansible 2.2 (https://github.com/ansible/ansible/issues/13981#issuecomment-247031451). I might want to add a more detailed Ansible version checker thingie, because the current one only checks once the entire playbook is loaded and validated—meaning I get a lot of issue reports for people running 2.1 and hitting the systemd error.